### DISCUSSION TITLE
Cannot prevent certvalidator from validating CRLs

### DISCUSSION THREAD

**Discussion by fredericoschardong** [MatthiasValvekens/pyHanko#381] [General]
Dear Matthias,

In `pyhanko-certvalidator` 0.26.0 this piece of code was introduced in `validate.intl_validate_path`:

```
    leaf_asserted_nonrevoked = False
    revinfo_manager = validation_context.revinfo_manager
    if isinstance(path.leaf, x509.Certificate):
        leaf_asserted_nonrevoked = revinfo_manager.check_asserted_unrevoked(
            path.leaf, moment
        )
```

This seems to be executed despite creating a fairly "do not touch the CRL" validation policy, e.g.:

```
    revinfo_policy = CertRevTrustPolicy(
        NO_REVOCATION,
        retroactive_revinfo=False,
    )

    vc = ValidationContext(trust_roots = trusted_roots, 
                        allow_fetching=False, 
                        revinfo_policy=revinfo_policy,
                        time_tolerance=timedelta(seconds=60))
```

I have a very specific use case where the CRL shouldn't be downloaded nor validated at all, and `check_asserted_unrevoked` is breaking things up. Is there something I can do to go around it or avoid it altogether?

**Comment by MatthiasValvekens** [comment]
Oh, that's odd. That's a feature intended for exempting specific certs from revocation checks while still checking everything else, but it probably shouldn't trigger if revocation checking is disabled altogether.

Can you share a stack trace?
